name: Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

permissions:
  contents: read

jobs:
  tests:
    name: tests
    runs-on: ubuntu-22.04
    concurrency:
      # https://github.community/t/concurrecy-not-work-for-push/183068/7
      group: ${{ github.event_name == 'push' && github.run_number || github.ref }}-tests
      cancel-in-progress: true
    steps:
    - name: Checkout
      uses: actions/checkout@v3

    - name: Install Python
      uses: actions/setup-python@v4
      with:
        python-version: '3.10'

    - name: Install dependencies
      run: |
        sudo apt install -y libxcb-xinerama0 pyqt5-dev-tools libxkbcommon-x11-0
        python -m pip install --upgrade pip
        pip install .[test] pytest-xvfb

    - name: Run tests
      env:
        QT_DEBUG_PLUGINS: 1
      run: |
        ulimit -c unlimited
        pytest tests --durations 20

    - name: chmod
      if: always()
      run: |
        if [[ -d "$DIRECTORY" ]]; then
        sudo chmod -R +rwx /var/crash
        fi

    - name: Upload artifacts
      if: always()
      uses: actions/upload-artifact@master
      with:
        name: crashes
        path: /var/crash
